expo/expo
# [docs] Refactor analytics into a provider

# Why

Part of [ENG-4086](https://linear.app/expo/issue/ENG-4086/refactor-analytics-and-group-the-implementation)

# How

This pulls the analytics out of both **common** and the **pages/_app.tsx**. Instead of having references in both files, its all contained within the **providers/Analytics.tsx** file. It does import/export some stuff from the **pages/_app.tsx** file. 

# Test Plan

- Inspect network tab to see if all scripts are loaded
- Inspect network tab to see if page views are reported

# Checklist
